src/endless/EndlessUsbToolDlg.cpp

```diff
@@ -4820,10 +4842,15 @@ bool CEndlessUsbToolDlg::CopyFilesToESP(const CString &fromFolder, const CString
 
 void CEndlessUsbToolDlg::ImageUnpackCallback(const uint64_t read_bytes)
 {
+    UpdateUnpackProgress(read_bytes, CEndlessUsbToolDlg::ImageUnpackFileSize);
+}
+
+void CEndlessUsbToolDlg::UpdateUnpackProgress(const uint64_t current_bytes, const uint64_t total_bytes)
```

Comment:
For SquashFS that's true, but  7z doesn't guarantee to report progress in *decompressed* bytes, it could be measured in bytes of the input file. I will add a comment.